src/products/hooks/useInitFetchProducts.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect } from "react";
+import { fetchProducts } from "../slice";
+import { selectProductsStatus } from "../slice";
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatch } from "../../_shared";
+
+export const useInitFetchProducts = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const status = useSelector(selectProductsStatus);
+
+  useEffect(() => {
+    if (status === "idle") {
+      dispatch(fetchProducts());
+    }
+  }, [status]);
```

Comment:
let's add the es-lint rule for the hooks, also dispatch should be added to the effect dependencies